housekeeper/store/filters/file_join_filters.py

```diff
@@ -0,0 +1,41 @@
+from enum import Enum
+from typing import Callable, List, Optional
+from sqlalchemy.orm import Query
+from sqlalchemy import func as sqlalchemy_func
+
+from housekeeper.store.models import File, Tag
+
+
+def filter_files_by_tags(files_query: Query, tag_names: List[str], **kwargs) -> Query:
```

Comment:
```suggestion
def filter_files_by_tags(files: Query, tag_names: List[str], **kwargs) -> Query:
```

I think in CG we are not explicit with the naming in this way (i.e. files_query) We just say files or samples. Maybe we should keep the same style here as well.